collections/_posts/2024-12-02-SSL_comparison.md

```diff
@@ -0,0 +1,123 @@
+| TMED-2      | Echocardiogram US images    | 4                  | 112*112    |
+| AIROGS      | Retina color fundus         | 2                  | 384*384    |
+
+The details of the images' distribution are in the following table, with the number of images labeled or not by category. The two first datasets are entirely labeled, so a part of the images were considered as unlabeled to fit the training paradigm of SSL.
+
+![](/collections/images/SSL_comparison/datasets_description.jpg)
+
+Here are some images examples.
+
+![](/collections/images/SSL_comparison/datasets_example.jpg)
+
+# Experimental setup
+
+For the low resolution images ResNet-18 is used, WideResNet-28-2 is used for TMED-2 and ResNet-18 or 50 are tested for AIROGS dataset. \
+A maximum of 200 epochs is performed and training is stopped if the balanced accuracy plateaus for 20 consecutive epochs. 
+
+Hyperparameters (learning rate, weight decay and unlabeled loss weight, in addition to the parameters specific for each method) optimization is made for each method and dataset. To fit with real life constraints, they fixed a certain number of hours with a NVIDIA A100 GPU for each hyperparameter optimization (25h for PathMNIST, 50h for TissueMNIST and 100h for TMED-2 and AIROGS). Within this fixed budget, a random search of hyperparameters is done, tacking the best with validation set.
+
+# Results
+
+Five separate trials were performed and mean balanced accuracy is calculated each 30 or 60 minutes on validation and test sets.
+
+Balanced accuracy on test set over time is represented for each dataset and method. As there is a rough monotonic improvement in test performance over time despite using a realistic-size validation set, they conclude that checkpoint selection and hyperparameters optimization can be effective with a realistically-sized validation set.
+
+![](/collections/images/SSL_comparison/accuracy_evolution.jpg)
+
+All the methods are compared and none of them clearly stands out from the other depending on the dataset. So they measure the balanced accuracy relative gain over the best supervised method for each dataset. MixMatch represent the best overall choice.
+
+![](/collections/images/SSL_comparison/results_comparison.jpg)
+
+The tuning strategy, even though it reserves data for a validation set, is competitive in comparison with transferring the hyperparameters from another dataset (natural images or medical images) as it may have been done in other benchmark papers.
+
+![](/collections/images/SSL_comparison/hyperparameters_strategy_evaluation.jpg)
+
+
+# Discussion
+
+There is a real gain in performance of performing SSL, but it depends on the dataset (less gain on TMED-2). However, they insist on the importance of the selection of the hyperparameters and proper evaluation protocol depending on specific needs. Especially MixMatch requires cautious hyperparameters tuning when applied to a new dataset.
```

Comment:
- "There is a real gain in performance of performing SSL" → Suggested: "SSL provides a tangible performance gain..."
- "less gain on TMED-2" → Unclear. Suggested: "a smaller performance improvement is observed for TMED-2."
- "MixMatch requires cautious hyperparameters tuning" → Suggested: "MixMatch requires careful hyperparameter tuning..."